### PR TITLE
Organize test structure, integrate go modules

### PR DESCRIPTION
**Pull request by aacanakin**

### CODE REVIEW

**Comment by aacanakin** [comment]
Refactor postgres, sqlite and mysql e2e tests as well

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/aacanakin/qb
```
Comment (cdevienne) [inline go.mod]: The main repo is still on slicebit/qb.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/go-sql-driver/mysql v1.4.1
+	github.com/jmoiron/sqlx v1.2.0
+	github.com/mattn/go-sqlite3 v1.10.0
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/serenize/snaker v0.0.0-20171204205717-a683aaf2d516
+	github.com/stretchr/testify v1.2.2
+)
```
Comment (cdevienne) [inline go.mod]: Why is there mysql & sqlite drivers but not postgresql ?

Comment (cdevienne) [reply go.mod]: Just saw the other commit, sorry.

**Comment by aacanakin** [comment]
Hi @cdevienne,

I see your point about using subtests. However, the current status of tests are horrible. They are really flaky. The reason I refactored the tests is because I'm planning a complete redesign. The new design will include your visitor implementation that is split into interfaces, table integration of visitor, package structure done correctly, and go idioms implemented correctly.

So, these tests are here for regressions that may appear on the new design.
Right now, testify's test suite & setup functions are enough for the project's needs. Setup function is called for each test case. I'll create an issue to evaluate & move to subtests.
I'll provide a release tag after this pr. This pr has no breaking change except glide. In your library, you can use the release tag.

**Comment by cdevienne** [comment]
The refactoring was long-needed, so it's definitely a good thing.
I am ok with postponing the switch to subtests

**Pull request merged**